homeassistant/components/wiz/light.py

```diff
@@ -0,0 +1,348 @@
+    # Add devices with defined name
+    async_add_entities([wizbulb], update_before_add=True)
+    return True
+
+
+class WizBulbEntity(LightEntity):
+    """Representation of WiZ Light bulb."""
+
+    def __init__(self, light: wizlight, name, scenes):
+        """Initialize an WiZLight."""
+        self._light = light
+        self._state = None
+        self._brightness = None
+        self._attr_name = name
+        self._rgb_color = None
+        self._temperature = None
+        self._hscolor = None
+        self._available = None
+        self._effect = None
+        self._scenes: list[str] = scenes
+        self._bulbtype: BulbType = light.bulbtype
+        self._mac = light.mac
+        self._attr_unique_id = light.mac
+        # new init states
+        self._attr_min_mireds = self.get_min_mireds()
+        self._attr_max_mireds = self.get_max_mireds()
+        self._attr_supported_features = self.get_supported_features()
+
+    @property
+    def brightness(self):
+        """Return the brightness of the light."""
+        return self._brightness
+
+    @property
+    def rgb_color(self):
+        """Return the color property."""
+        return self._rgb_color
+
+    @property
+    def hs_color(self):
+        """Return the hs color value."""
+        return self._hscolor
+
+    @property
+    def is_on(self):
+        """Return true if light is on."""
+        return self._state
+
+    async def async_turn_on(self, **kwargs):
+        """Instruct the light to turn on."""
+        brightness = None
+
+        if ATTR_BRIGHTNESS in kwargs:
+            brightness = kwargs.get(ATTR_BRIGHTNESS)
+
+        if ATTR_RGB_COLOR in kwargs:
+            pilot = PilotBuilder(rgb=kwargs.get(ATTR_RGB_COLOR), brightness=brightness)
+
+        if ATTR_HS_COLOR in kwargs:
+            pilot = PilotBuilder(
+                hucolor=(kwargs[ATTR_HS_COLOR][0], kwargs[ATTR_HS_COLOR][1]),
+                brightness=brightness,
+            )
+        else:
+            colortemp = None
+            if ATTR_COLOR_TEMP in kwargs:
+                kelvin = color_utils.color_temperature_mired_to_kelvin(
+                    kwargs[ATTR_COLOR_TEMP]
+                )
+                colortemp = kelvin
+                _LOGGER.debug(
+                    "[wizlight %s] kelvin changed and send to bulb: %s",
+                    self._light.ip,
+                    colortemp,
+                )
+
+            sceneid = None
+            if ATTR_EFFECT in kwargs:
+                sceneid = get_id_from_scene_name(kwargs[ATTR_EFFECT])
+
+            if sceneid == 1000:  # rhythm
+                pilot = PilotBuilder()
+            else:
+                pilot = PilotBuilder(
+                    brightness=brightness, colortemp=colortemp, scene=sceneid
+                )
+                _LOGGER.debug(
+                    "[wizlight %s] Pilot will be send with brightness=%s, colortemp=%s, scene=%s",
```

Comment:
will be sent

Reply:
https://github.com/home-assistant/core/pull/65819